cluster_experiments/analysis_plan.py

```diff
@@ -0,0 +1,264 @@
+from typing import List
+
+import pandas as pd
+from pandas import DataFrame
+
+from cluster_experiments.analysis_results import (
+    AnalysisPlanResults,
+    HypothesisTestResults,
+)
+from cluster_experiments.hypothesis_test import HypothesisTest
+from cluster_experiments.power_config import analysis_mapping
+from cluster_experiments.variant import Variant
+
+
+class AnalysisPlan:
+    """
+    A class used to represent an Analysis Plan with a list of hypothesis tests and a list of variants.
+
+    Attributes
+    ----------
+    tests : List[HypothesisTest]
+        A list of HypothesisTest instances
+    variants : List[Variant]
+        A list of Variant instances
+    variant_col : str
+        name of the column with the experiment groups
+    alpha : float
+        significance level used to construct confidence intervals
+
+    Methods
+    -------
+    __init__(self, tests: List[HypothesisTest], variants: List[Variant]):
+        Initializes the AnalysisPlan with the provided list of hypothesis tests and variants.
+    _validate_inputs(tests: List[HypothesisTest], variants: List[Variant]):
+        Validates the inputs for the AnalysisPlan class.
+    """
+
+    def __init__(
+        self,
+        tests: List[HypothesisTest],
+        variants: List[Variant],
+        variant_col: str = "treatment",
+        alpha: float = 0.05,
+    ):
+        """
+        Parameters
+        ----------
+        tests : List[HypothesisTest]
+            A list of HypothesisTest instances
+        variants : List[Variant]
+            A list of Variant instances
+        variant_col : str
+            The name of the column containing the variant names.
+        alpha : float
+            significance level used to construct confidence intervals
+        """
+        self._validate_inputs(tests, variants, variant_col)
```

Comment:
wdyt is best, doing this here or a self._validate_inputs() after seeting the attributes in the class?

Reply:
I think here as it can be flagged sooner no?